### ISSUE TITLE
receive and forward image or video?

### ISSUE THREAD

**Issue by NabiilahDatabase** [vasani-arpit/WBOT#103]
I am using the method:
WAPI.waitForNewMessage(cb);

When i get text everthing is great,
when i get image or video, the base64 is as thumbnail.

How can i get the original image?

I also see that i have addition fields:
clientUrl: "https://mmg-fna.whatsapp.net/d/f/AuZC-stp7PugSORvrhcGGEF6gYfXix_NkYLGx9zaz05L.enc",

directPath: "/v/t62.7118-24/19481306_492010328088217_6362975681879656851_n.enc?
oh=f2298e63de52ac5f1a7cba28a6d13fd1&oe=C3CE5E9F",
filehash: "BGf0teAJnNXfrv/3xyVc5P9/Q9b2QrKKtYKBPbJOxtw=",
uploadhash: "jjO6D54mBIz9puNx2Ha2ESHYbCC2c1iA3pZN4POEPBQ=",
mediaKey: "04BhnmNBWL0xqjt9tDEquv+BmGfA1yvuQsU8rQOSwCg="

Is there any helper method?

**Comment by vasani-arpit** [comment]
You can use `WAPI.downloadFile(msg.clientUrl)` to download the image.

**Comment by didom900** [comment]
> You can use `WAPI.downloadFile(msg.clientUrl)` to download the image.

Arpit, then with blob, how do you convert it to file, I have tried many things but when I'm going to open the file whoe me the file is damage.

**Comment by igoeneo** [comment]
This is how I download media file through webhook.

Add/Modify in WAPI.js

```javascript
window.WAPI.downloadFile2 = async function(url, done) {
    let xhr = new XMLHttpRequest();
    xhr.onload = function () {
        if (xhr.readyState == 4) {
            if (xhr.status == 200) {
                done(xhr.response);
            } else {
                console.error(xhr.statusText);
            }
        } else {
            console.log(err);
            done(false);
        }
    };
    xhr.open("GET", url, true);
    xhr.responseType = 'arraybuffer';
    //xhr.responseType = 'blob';
    xhr.send(null);

};
```
Add identifier for your media file in inject.js and prepare for sending it to your webhook. Actually I add this for media files only. If its text, your "else" is the original one.
```javascript
body.caption = message.caption; // This is for filename, for image and video you must add caption (filename.jpg). 
// PDF, words, excel, ppt, etc not needed to add caption, it will generated itself from its filename.

if(message.isMMS){// This is Boolean, so if it's media file return true.
   WAPI.downloadFile2(message.clientUrl, function(result) {
      Store.CryptoLib.decryptE2EMedia(message.type.toLowerCase(), result, message.mediaKey, message.mimetype)
	  .then(function(a) {
         let reader = new FileReader();
         reader.onload = function (e) {
            window.log("reader.result is" + reader.result); //for checking, is it working
            body.text = reader.result;// result is data:*/*;base64,blablablablablabla
            var fmtText = {
               method: "POST",
               body: JSON.stringify(body),
               headers: {
                  'Content-Type': 'application/json'
               }
            }
            if (intents.appconfig.webhook) {
               fetch(intents.appconfig.webhook, fmtText)
               .then((resp) => resp.json()).then(function (response) {
                  //response received from server
                  console.log(response);
                  WAPI.sendSeen(message.chatId._serialized);
                  //replying to the user based on response
                  if (response && response.length > 0) {
                     response.forEach(itemResponse => {
                        WAPI.sendMessage2(message.chatId._serialized, itemResponse.text);
                        //sending files if there is any 
                        if (itemResponse.files && itemResponse.files.length > 0) {
                           itemResponse.files.forEach((itemFile) => {
                              window.getFile(itemFile.file).then((base64Data) => {
                                 //console.log(file);
                                 WAPI.sendImage(base64Data, message.chatId._serialized, itemFile.file);
                              }).catch((error) => {
                                 window.log("Error in sending file\n" + error);
                              })
                           })
                        }
                     });
                  }
               }).catch(function (error) {
                  console.log(error);
               });
            }
         };
         reader.readAsDataURL(a._blob);
      }).catch(function(e) {
         window.log("Error: " + e);
      });
   });
}
else{
	 //body.original = message;
	 body.text = message.body;
	 var fmtText = {
		method: "POST",
		body: JSON.stringify(body),
		headers: {
		   'Content-Type': 'application/json'
		}
	 }
	 if (intents.appconfig.webhook) {
		fetch(intents.appconfig.webhook, fmtText)
		.then((resp) => resp.json()).then(function (response) {
		   //response received from server
		   console.log(response);
////////////////////////////////////////////////////// and so on.
```

Receive it in webhook on index.js with expressjs 
```javascript
async function (req, res, next){
  var check = req.body.text.split(":");//split for  checking if it's media file
  if(check[0].toLowerCase() == 'data'){
	 var wftl = check[1].split(",");//get base64 data
	 var pathDownload = "X:\\Shared drives\\Probo's\\wbot\\";
	 var fileDownload = pathDownload + req.body.caption ;
	 require('fs').writeFile(fileDownload, wftl[1], 'base64', function(err) {
		 if (err) throw err;
		console.log('Download!');
	 });
	 res.send([{
	   "text": req.body.caption + " downloaded!",
	   "type": "message"
	 }])
  }
  else next()
}
```

Hope this will help.

**Comment by jasp402** [comment]
> > You can use `WAPI.downloadFile(msg.clientUrl)` to download the image.
> 
> Arpit, then with blob, how do you convert it to file, I have tried many things but when I'm going to open the file whoe me the file is damage.

I have the same problem

**Comment by vasani-arpit** [comment]
This has been resolved in v0.17.0
